### DISCUSSION TITLE
Why don't I get any shrinks?

### DISCUSSION THREAD

**Discussion by exarkun** [dubzzz/fast-check#4956] [Q&A] [answered]
I'm using fast-check (3.15.1) with mocha (10.2) and chai (4.3.10) and I can't get the runner to ever try any shrinks.  I see this with my semi-complex arbitraries (whether they use chain or not) but I also get see it for trivia uses like this one:

```
describe("foo", () => {
  it("bar", () => {
        fc.assert(
        fc.property(
          fc.integer(),
          (n: number) => {
            expect(n).to.deep.equal(3);
          },
        ),
      );
  });
});
```

The output from running this:

```
Property failed after 1 tests
{ seed: -2071887809, path: "0", endOnFailure: true }
Counterexample: [-1471789753]
Shrunk 0 time(s)
Got AssertionError: expected -1471789753 to deeply equal 3
```

Surely `-1471789753` is worth trying to shrink a bit?  I feel I must be missing something.

**Comment by gruhn** [comment]
I tried to reproduce this issue with the exact packages you mention (see: https://github.com/gruhn/fast-check-issue-4956/tree/master) but I get proper shrinking:
```
Property failed after 1 tests
{ seed: -346953532, path: "0:0", endOnFailure: true }
Counterexample: [0]
Shrunk 1 time(s)
Got AssertionError: expected +0 to deeply equal 3
```
Maybe pull that repository and compare.

**Comment by dubzzz** [comment, marked as answer]
Do you happen to have a global configuration of fast-check preventing it to shrink on failure? Something done via `fc.confgureGlobal(...)`.

**Reply by exarkun** [reply]
Ah, yes.  The project has global config to set `endOnFailure` to `true` :/  If I remove that setting, my example does shrink so that seems to be the explanation, thank you.  There is one piece of this I don't quite understand though.  If I run the test with `fc.configureGlobal({ endOnFailure: true })` or `fc.configureGlobal({ endOnFailure: false })` or even with the `endOnFailure` customization deleted entirely from that argument object, when the test fails the report includes:

```
  Error: Property failed after 1 tests
  { seed: 1895761743, path: "0:0", endOnFailure: true }
```

Note **endOnFailure: true** - and in gruhn's output, I see this is also the case.  Verging on a different question now, but how should this part of the output be interpreted?

**Reply by dubzzz** [reply]
In case of failure fast-check suggests you to focus only on the failure. In order to do so it gives you the line: `{ seed: 1895761743, path: "0:0", endOnFailure: true }`. This line is just a way to only have that test running and no other. It avoids fast-check to run this test, then all the shrinks... It's just for convenience.

**Reply by exarkun** [reply]
Ah, that makes sense.  Thanks again.
